File: Source/UnitTests/Translator/IntegrationTest/DotNetExpected/AbstractClass.cs
namespace Test.Integration
{
	public abstract class AbstractClass : Interface
	{
		private string name;
		public string Name {

			get { return this.name; }
			set { this.name = value; }
		}

		protected bool existSimilarFieldAndMethod = true;
		protected internal bool ExistSimilarFieldAndMethod()
		{
			string methodName = Name;
			Name = methodName;
			return existSimilarFieldAndMethod;
		}

		public abstract void AbstractClassMethod();

		public abstract class InnerAbstractClass
		{
			public InnerAbstractClass(int num)
			{
			}
			public abstract void InnerAbstractClassMethod(int num);
		}
		public abstract string InterfaceMethod1(string _string);
	}
}

File: Source/UnitTests/Translator/IntegrationTest/Base/AbstractClass.java
package Test.Integration;
abstract class AbstractClass implements Interface
{
	private String name;

	public void setName(String name)
	{
		this.name = name;
	}
	public String getName()
	{
		return this.name;
	}

	protected final boolean existSimilarFieldAndMethod = true;
	protected boolean existSimilarFieldAndMethod()
    {
		String methodName = getName();
		setName(methodName);
        return existSimilarFieldAndMethod;
    }

	public abstract void AbstractClassMethod() throws Exception;

	public abstract static class InnerAbstractClass
	{
		public InnerAbstractClass(int num)
		{
        }
		public abstract void InnerAbstractClassMethod(int num);
	}
	
	public void methodToExclude()
	{
	}
}
